### PR TITLE
Update go to 1.19.2

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does
This PR updates Go version in build image to 1.19.2

#### Checklist

- [na] Tests updated
- [na] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pracucci — COMMENTED** [review]
Could you publish and replace the build image, as well as change the version in `.github/workflows/test-build-deploy.yml` too, please?

**Review by aknuds1 — DISMISSED** [review]
The changelog also needs updating, in addition to what Marco mentioned.

**Comment by pstibrany** [comment]
> Could you publish and replace the build image, as well as change the version in .github/workflows/test-build-deploy.yml too, please?

Still building, this was meant to be draft. Oops

**Comment by pstibrany** [comment]
This is ready for review.

File: .github/workflows/test-build-deploy.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
   lint:
```
Comment (pracucci) [inline .github/workflows/test-build-deploy.yml:20]: Look for "1.19" in this file. There's an entry to update.

**Review by pracucci — APPROVED** [review]
LGTM (modulo a comment)

**Review by aknuds1 — COMMENTED** [review]
There's still one 1.19.1 reference in .github/workflows/test-build-deploy.yml.

**Comment by pstibrany** [comment]
Thanks, fixed reference to 1.19.1 in integration tests setup.

**Comment by pstibrany** [comment]
Unfortunately this doesn't help with the regexp problem, because Mimir is using fork of regexp library, that's available at github.com/grafana/regexp (`speedup` branch). We need to fix the problem there first.

**Comment by pstibrany** [comment]
Here is updated speedup branch: https://github.com/grafana/regexp/pull/2.

**Pull request merged**